File: typescript/src/main/vsdk/toolkit/common/statistics/RadianceStatistics.ts
import { ColorRgb } from "../ColorRgb";

export class RadianceStatistics {
  public totalArea: number;
  public maxSelfEmittedRadiance: ColorRgb;
  public maxSelfEmittedPower: ColorRgb;
  public referenceLuminance: number;
  public totalEmittedPower: ColorRgb;
  public estimatedAverageRadiance: ColorRgb;
  public averageReflectivity: ColorRgb;

  public constructor() {
    this.totalArea = 0.0;
    this.maxSelfEmittedRadiance = new ColorRgb();
    this.maxSelfEmittedPower = new ColorRgb();
    this.referenceLuminance = 0.0;
    this.totalEmittedPower = new ColorRgb();
    this.estimatedAverageRadiance = new ColorRgb();
    this.averageReflectivity = new ColorRgb();
  }

  public reset(): void {
    this.totalArea = 0.0;
    this.maxSelfEmittedRadiance.clear();
    this.maxSelfEmittedPower.clear();
    this.referenceLuminance = 0.0;
    this.totalEmittedPower.clear();
    this.estimatedAverageRadiance.clear();
    this.averageReflectivity.clear();
  }
}

File: java/src/main/java/vsdk/toolkit/common/statistics/RadianceStatistics.java
package vsdk.toolkit.common.statistics;

import vsdk.toolkit.common.color.ColorRgb;

public class RadianceStatistics {
    public float totalArea;
    public ColorRgb maxSelfEmittedRadiance;
    public ColorRgb maxSelfEmittedPower;
    public double referenceLuminance;
    public ColorRgb totalEmittedPower;
    public ColorRgb estimatedAverageRadiance;
    public ColorRgb averageReflectivity;

    public RadianceStatistics() {
        totalArea = 0.0f;
        maxSelfEmittedRadiance = new ColorRgb();
        maxSelfEmittedPower = new ColorRgb();
        referenceLuminance = 0.0;
        totalEmittedPower = new ColorRgb();
        estimatedAverageRadiance = new ColorRgb();
        averageReflectivity = new ColorRgb();
    }

    public void reset() {
        totalArea = 0.0f;
        maxSelfEmittedRadiance.clear();
        maxSelfEmittedPower.clear();
        referenceLuminance = 0.0;
        totalEmittedPower.clear();
        estimatedAverageRadiance.clear();
        averageReflectivity.clear();
    }
}
